File: src/StorageService/StorageService.Domain/Entities/Detail.cs
using StorageService.Domain.Primitives;

namespace StorageService.Domain.Entities;

public class Detail : Entity
{
    public string NomenclatureCode { get; set; }
    public string Name { get; set; }
    public int Count { get; set; }
    public int StorekeeperId { get; set; }
    public Storekeeper Storekeeper { get; set; }
    public bool IsDeleted { get; set; }
    public DateTime CreatedAtDate { get; set; }
    public DateTime? DeletedAtDate { get; set; }
}

File: src/storage-service-client/src/app/core/models/details/detail.ts
import {Storekeeper} from '../storekeepers/storekeeper';

export interface Detail {
  id: number;
  nomenclatureCode: string;
  name: string;
  count: number;
  storekeeperId: number;
  storekeeper: Storekeeper | null;
  isDeleted: boolean;
  createdAtDate: string;
  deletedAtDate: string | null;
}
